src/lib/router.js

```diff
@@ -11,6 +11,9 @@ const domparser = new DOMParser();
  */
 async function getPage(url) {
   const res = await fetch(`/${url}`);
+  if (!res.ok) {
```

Comment:
yeah it seems like we should render a 404 template in this scenario.

Reply:
I think this is more part of serving infra itself. I'm happy with this change for now because loading a broken page didn't really work properly—and generated a pageview that looked real—although I confess it's reasonably orthogonal from the Analytics change.

Reply:
oh sorry, I didn't mean we had to add the 404 stuff to this particular PR. Just wanted to mention it.